homeassistant/components/forked_daapd/media_player.py

```diff
@@ -0,0 +1,695 @@
+        await _async_setup_platform(
+            hass, config=config_entry.data, async_add_entities=async_add_entities
+        )
+        await update_listener(hass, config_entry)
+        config_entry.add_update_listener(update_listener)
+
+
+async def update_listener(hass, entry):
+    """Handle options update."""
+    hass.data[DOMAIN].update_options(entry.options)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the forked-daapd platform.
+
+    Deprecated.
+    """
+    _LOGGER.warning(
+        "Setting configuration for forked-daapd via platform is deprecated. "
+        "Configure via the forked-daapd integration instead."
+    )
+    await _async_setup_platform(hass, config, async_add_entities, discovery_info)
+
+
+async def _async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the forked-daapd platform."""
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    password = config.get(CONF_PASSWORD)
+    default_volume = config.get(CONF_DEFAULT_VOLUME)
+    forked_daapd_device = ForkedDaapdDevice(
+        hass=hass,
+        name=name,
+        ip_address=host,
+        api_port=port,
+        api_password=password,
+        default_volume=default_volume,
+        async_add_entities=async_add_entities,
+    )
+    await forked_daapd_device.async_init()  # updates essential info and adds entities
+    hass.data[DOMAIN] = forked_daapd_device
+
+
+class ForkedDaapdZone(MediaPlayerDevice):
+    """Representation of a forked-daapd output."""
+
+    @property
+    def unique_id(self):
+        """Return unique ID."""
+        return self._output_id
+
+    @property
+    def should_poll(self) -> bool:
+        """Entity pushes its state to HA."""
+        return False
+
+    async def async_toggle(self):
+        """Toggle the power on the zone.
+
+        Default media player component method counts idle as off - we consider idle to be on but just not playing.
+        """
+        if self.state == STATE_OFF:
+            await self.async_turn_on()
+        else:
+            await self.async_turn_off()
+
+    def __init__(self, master_device, output, default_volume):
+        """Initialize the ForkedDaapd Zone."""
+        super(ForkedDaapdZone, self).__init__()
+        _LOGGER.debug("New ForkedDaapdZone")
+        self._master = master_device
+        self._name = output["name"]
+        self._type = output["type"]
+        self._output_id = output["id"]
+        self._last_volume = (
+            output["volume"] if output["volume"] else default_volume
+        )  # used for mute/unmute
+        self._available = True
+
+    @property
+    def available(self) -> bool:
+        """Return whether the zone is available."""
+        return self._available
+
+    @available.setter
+    def available(self, val):
+        """Setter for property."""
+        self._available = val
+
```

Comment:
Ok. Should I just remove the getter here too and rename the _available variable to available to allow direct access?

Reply:
No we need to keep the `available` property since it's part of the base entity class. Make a private attribute `self._available` that we return for that property and that we can change as needed. The change should come from inside the entity. If we need to do it from outside the entity we use our dispatch helper signals for that.